util/dnsname/dnsname.go

```diff
@@ -21,31 +21,40 @@ const (
 type FQDN string
 
 func ToFQDN(s string) (FQDN, error) {
-	if isValidFQDN(s) {
-		return FQDN(s), nil
-	}
 	if len(s) == 0 || s == "." {
 		return FQDN("."), nil
 	}
 
-	if s[len(s)-1] == '.' {
-		s = s[:len(s)-1]
-	}
 	if s[0] == '.' {
 		s = s[1:]
 	}
-	if len(s) > maxNameLength {
+	raw := s
+	totalLen := len(s)
+	if s[len(s)-1] == '.' {
```

Comment:
Slight pref for the more explicit:

```go
     s = strings.TrimSuffix(s, ".")
     if s != raw {
           totalLen != 1 // a dot was removed
     }
```

Might be nanoseconds slower, but still doesn't allocate, which is the main goal here.

Reply:
Done. Happy path is 30% slower than before this PR, but it's just 9ns vs. 12ns.